File: TP2/Exo 2.4/python/concat.py
def concat(chaine1, chaine2):
    # Concaténation de deux chaînes
    result = chaine1 + chaine2  # Combine les deux chaînes
    return result

def main():
    # Demander à l'utilisateur de saisir deux chaînes de caractères
    chaine1 = input("Saisissez la première chaîne de caractères : ")
    chaine2 = input("Saisissez la deuxième chaîne de caractères : ")
    
    # Concaténer les deux chaînes
    concat_result = concat(chaine1, chaine2)
    
    # Afficher le résultat de la concaténation
    print(f"Concaténation des deux chaînes : {concat_result}")

# Point d'entrée du programme
if __name__ == "__main__":
    main()

File: TP2/Exo 2.4/C/concat.c
#include <stdio.h>

// Fonction qui concatène la chaîne source à la fin de la chaîne destination
void concat(char destination[], char source[])
{
    int i = 0, j = 0;

    // Trouve la fin de la chaîne destination
    while (destination[i] != '\0')
    {
        i++;
    }

    // Copie les caractères de la source à partir de la fin de la destination
    while (source[j] != '\0')
    {
        destination[i] = source[j];
        i++;
        j++;
    }

    destination[i] = '\0'; // Ajoute le caractère nul à la fin de la destination
}

void carac(char destination[], char source[])
{
    concat(destination, source); // Concaténer la source dans la destination
    printf("Chaine finale : %s\n", destination); // Affiche la chaîne finale complète
}

int main()
{
    char chaine1[200]; // Allouer suffisamment de place pour la chaîne finale
    char chaine2[100]; // Allouer suffisamment de place pour la chaîne source

    printf("Envoie la première chaine de caractères (Destination) : ");
    scanf("%199s", chaine1); // Lire une chaîne de caractères jusqu'à 199 caractères

    printf("Envoie la deuxième chaine de caractères (Source) : ");
    scanf("%99s", chaine2); // Lire une chaîne de caractères jusqu'à 99 caractères

    carac(chaine1, chaine2); // Appeler la fonction pour concaténer les chaînes
    
    return 0;
}
